packages/frontend/src/components/common/Dashboard/DashboardHeader.tsx

```diff
@@ -180,6 +180,34 @@ const DashboardHeader = ({
         }),
     );
 
+    const handleDashboardRefreshUpdateEvent = useCallback(
+        (intervalMin?: number) => {
+            track({
+                name: EventName.DASHBOARD_AUTO_REFRESH_UPDATED,
+                properties: {
+                    useId: user.data?.userUuid,
+                    dashboardId: dashboardUuid,
+                    organizationId: organizationUuid,
+                    projectId: projectUuid,
+                    frequency: intervalMin ? `${intervalMin} minutes` : 'off',
+                },
+            });
+        },
+        [
+            dashboardUuid,
+            organizationUuid,
+            projectUuid,
+            track,
+            user.data?.userUuid,
+        ],
+    );
+
+    useEffect(() => {
+        // When unmounting the dashboard - navigate away - track the event with undefined interval
+        // Not doing it inside the button component because it get's triggered too often
+        return () => handleDashboardRefreshUpdateEvent(undefined);
```

Comment:
🤔 I think we should check if unmounted + if there was a refresh value previously...

Also, have you tried wrapping `DashboardRefreshButton` with `memo()`? I believe that reduces the unmounts and also allows you to move this to inside the component.

Reply:
I ended up fixing this in the last commit by having a memo variable called `hasInterval` since the interval changing was what was triggering the undefined call of `onIntervalChange`.
Added the memo but at least by itself didn't help a lot because of the above.

Before:
![Screenshot 2024-07-03 at 12 07 12](https://github.com/lightdash/lightdash/assets/22939015/76900608-231d-4ec1-96fe-10d484f5a17c)

After:
![Screenshot 2024-07-03 at 12 07 44](https://github.com/lightdash/lightdash/assets/22939015/a587e900-f12b-4809-8db0-da5c3a0b5bff)


These are for setting intervals and then navigating away from the dashboard 🎉